### ISSUE TITLE
Strange heatmap behavior when handling missing (NA) values

### ISSUE THREAD

**Issue by alkaZeltser** [uclahs-cds/package-BoutrosLab-plotting-general#176]
I'm working with `create.heatmap` and encountering odd behavior when dealing with NA values and single-row heatmaps.
The weirdest thing is that the NA missing values seem to be contaminating adjacent cells in certain cases.

I will have to resort to manually replacing NA with a pre-defined color as opposed to using the `fill.colour` option, which is kind of a waste of a nice feature.

For example:

```
real.continuous.test.data <-  t(data.frame(
    continuous.data = c(NA, '#3FDADC', '#BFF2F3', '#1FD4D6', NA,'#7FE6E8', '#5FE0E2', '#DFF8F9', '#FFFFFF', '#9FECED')
    ));

categorical.test.data <- t(data.frame(
    categorical.data = c(NA, 'red', 'blue', 'red', NA, 'green', 'yellow', 'purple', 'orange', 'pink')
    ));

# Ex1 Second column inherits NA from first column but the other NA is handled fine?
create.heatmap(
    x = real.continuous.test.data,
    input.colours = TRUE,
    clustering.method = 'none',
    same.as.matrix = TRUE,
    print.colour.key = FALSE
    );

# Ex2 Both second and fourth columns inherit NA from first and fifth column?
create.heatmap(
    x = categorical.test.data,
    input.colours = TRUE,
    clustering.method = 'none',
    same.as.matrix = TRUE,
    print.colour.key = FALSE
    )

# Ex3 Two rows, now the categorical data looks correct but continuous data remains contaminated
create.heatmap(
    x = rbind(categorical.test.data, real.continuous.test.data),
    input.colours = TRUE,
    clustering.method = 'none',
    same.as.matrix = TRUE,
    print.colour.key = FALSE
    )

```

Ex1:
<img width="300" alt="Screenshot 2024-04-06 at 12 50 11 AM" src="https://github.com/uclahs-cds/package-BoutrosLab-plotting-general/assets/30507223/a2dc9447-01c5-4cb0-9be8-0668215ff48c">

Ex2:
<img width="300" alt="Screenshot 2024-04-06 at 12 51 04 AM" src="https://github.com/uclahs-cds/package-BoutrosLab-plotting-general/assets/30507223/d94e7136-0ec2-45fa-b7ef-e1e70f3702c6">

Ex3:
<img width="300" alt="Screenshot 2024-04-06 at 12 51 52 AM" src="https://github.com/uclahs-cds/package-BoutrosLab-plotting-general/assets/30507223/ada0c1d0-4344-4867-b307-af5fb8f2f6e2">

**Comment by jarbet** [comment]
@alkaZeltser I didn't see your issue, but I posted a similar issue recently with a potential solution #181

**Comment by alkaZeltser** [comment]
> @alkaZeltser I didn't see your issue, but I posted a similar issue recently with a potential solution #181
Yeah I guess I already mentioned this workaround above, which is to replace NAs explicitly with colors. It's just unfortunate because the entire point of the fill.colour parameter is to generalize away this extra step.
